lightningd/lightningd.c

```diff
@@ -316,6 +318,7 @@ int main(int argc, char *argv[])
 
 	tal_free(ld);
 	opt_free_table();
+	tal_free(ctx);
```

Comment:
So, children are always freed before the parent, so the way I'd normally do this is to either have log be the parent of ld (which would look a bit weird), or make log's parent NULL and have ld's destructor free it.

But this works fine too.